modules/uhttp/filters.go

```diff
@@ -48,76 +48,83 @@ func (f FilterFunc) Apply(ctx fx.Context, w http.ResponseWriter, r *http.Request
 	f(ctx, w, r, next)
 }
 
-// contextFilter updates context to fx.Context
-func contextFilter(host service.Host) FilterFunc {
-	return func(ctx context.Context, w http.ResponseWriter, r *http.Request, next Handler) {
-		fxctx := fxcontext.New(ctx, host)
-		next.ServeHTTP(fxctx, w, r)
-	}
+type tracingServerFilter struct {
+	scope tally.Scope
 }
 
-func tracingServerFilter(host service.Host) FilterFunc {
-	return func(ctx context.Context, w http.ResponseWriter, r *http.Request, next Handler) {
-		// TODO:(anup) GFM-257 benchmark performance comparing with using type assertion
-		fxctx := &fxcontext.Context{
-			Context: ctx,
-		}
-		operationName := r.Method
-		carrier := opentracing.HTTPHeadersCarrier(r.Header)
-		spanCtx, err := opentracing.GlobalTracer().Extract(opentracing.HTTPHeaders, carrier)
-		if err != nil && err != opentracing.ErrSpanContextNotFound {
-			fxctx.Logger().Info("Malformed inbound tracing context: ", "error", err.Error())
-		}
-		span := opentracing.GlobalTracer().StartSpan(operationName, ext.RPCServerOption(spanCtx))
-		ext.HTTPUrl.Set(span, r.URL.String())
-		defer func() {
-			span.Finish()
-			fxctx.Logger().Debug("Span finished")
-		}()
-
-		fxctx = fxctx.WithContextAwareLogger(span)
-		fxctx = &fxcontext.Context{
-			Context: opentracing.ContextWithSpan(fxctx, span),
-		}
-		r = r.WithContext(fxctx)
-		next.ServeHTTP(fxctx, w, r)
+func (f tracingServerFilter) Apply(ctx fx.Context, w http.ResponseWriter, r *http.Request, next Handler) {
+	stopWatch := f.scope.Timer("tracing").Start()
```

Comment:
I added for auth because even if its a library call, we would want to understand the performance over time. I think I'll remove it for rest of the filters.